symengine/cwrapper.cpp

```diff
@@ -41,6 +41,8 @@ extern "C" {
 
 struct CRCPBasic {
     SymEngine::RCP<const SymEngine::Basic> m;
+    CRCPBasic() {};
```

Comment:
You don't need this anymore. I think this is what is causing the error, because an RCP object can't be null in Teuchos